### PR TITLE
feat(error): add a global ErrorBoundary component

### PR DESCRIPTION
**Pull request by charjac**
According to this: https://reactjs.org/blog/2017/07/26/error-handling-in-react-16.html
when a component throw an error it should be remove from the tree.

#287

### CODE REVIEW

File: app/containers/ErrorBoundary/index.js [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+class ErrorBoundary extends React.Component {
+  state = initialState;
+
+  static getDerivedStateFromError() {
+    return { hasError: true };
+  }
+
+  componentDidCatch(error) {
+    logger.error(error);
+    this.props.actions.error('error', 'Something wrong happen');
```
Comment (nukeop) [inline app/containers/ErrorBoundary/index.js]: happen**ed**

**Comment by nukeop** [comment]
Nice, I think this will work way better for unexpected errors than what happens now.
We could also use electron-timber (already in use in some components) to log these, if possible. Can you try adding that?

File: app/containers/ErrorBoundary/index.js (hunk 1 above)
Comment (charjac) [reply app/containers/ErrorBoundary/index.js]: done ! sorry for this i am french and we are terrible in foreign language :)

**Comment by charjac** [comment]
I already use electron-timber in the componentDidCatch hook l20 of ErrorBoundary/index.js

**Comment by nukeop** [comment]
Right, I missed that completely.

**Pull request merged**